zerver/webhooks/facebook/view.py

```diff
@@ -0,0 +1,114 @@
+                                page_id=payload['entry'][0]['changes'][0]['value']['page_id'],
+                                thread_id=payload['entry'][0]['changes'][0]['value']['thread_id']
+                            )
+    elif field == 'website':
+        message = message + 'Changed **website**'
+    return message
+
+def permissions_event(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry'][0]['changes'][0]['field']
+    message = '**{field} permission** changed'.format(field=field)
+    if field == 'ads_management':
+        message = message + '\n* {verb}'.format(
+            verb=payload['entry'][0]['changes'][0]['value']['verb']
+        )
+        for id in payload['entry'][0]['changes'][0]['value']['target_ids']:
+            message = message + '\n  * {id}'.format(id=id)
+    elif field == 'manage_pages':
+        message = message + '\n* {verb}'.format(
+            verb=payload['entry'][0]['changes'][0]['value']['verb']
+        )
+        for id in payload['entry'][0]['changes'][0]['value']['target_ids']:
+            message = message + '\n  * {id}'.format(id=id)
+    return message
+
+def application_event(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry'][0]['changes'][0]['field']
+    message = '**{field}** received'.format(field=field)
+    if field == 'plugin_comment':
+        message = message + '\n**{msg_user}:**\n```quote\n{message}\n```'.format(
+            msg_user=payload['entry'][0]['changes'][0]['value']['from']['name'],
+            message=payload['entry'][0]['changes'][0]['value']['message']
+        )
+    if field == 'plugin_comment_reply':
+        message = message + '\n**{prt_msg_user}:** (Parent)\n'\
+            '```quote\n{prt_message}\n```'.format(
+                prt_msg_user=payload['entry'][0]['changes'][0]['value']['parent']['from']['name'],
+                prt_message=payload['entry'][0]['changes'][0]['value']['parent']['message']
+            )
+        message = message + '\n**{cld_msg_user}:**\n```quote\n'\
+            '```quote\n{cld_message}\n```\n```'.format(
+                cld_msg_user=payload['entry'][0]['changes'][0]['value']['from']['name'],
+                cld_message=payload['entry'][0]['changes'][0]['value']['message']
+            )
+    return message
+
+@api_key_only_webhook_view("Facebook")
+@has_request_variables
+def api_facebook_webhook(request: HttpRequest, user_profile: UserProfile,
+                         stream: Text=REQ(default='Facebook'), token: Text=REQ()) -> HttpResponse:
+
+    if request.method == 'GET':  # facebook webhook verify
+        if request.GET.get("hub.mode") == 'subscribe':
+            if request.GET.get('hub.verify_token') == token:
+                return HttpResponse(request.GET.get('hub.challenge'))
+            else:
+                return json_error(('Error: Token is wrong'))
+        else:
+            return json_error(('Error: Unsupported method'))
+    else:  # facebook webhook receiver
```

Comment:
It looks like this `else` statement isn't required because all of the code paths above return; can you clean that up?  I think there's some similar unnecessary `else return` blocks above, too.